### PR TITLE
Adds LMDB as option for conductor

### PR DESCRIPTION
**Pull request by willemolding**
## PR summary

This is now the default persistence option.

This requires the holochain_persistence_* crates to be published and the versions bumped before a merge.

## testing/benchmarking notes

Tests will all fail until the creates are published

## changelog

Please check one of the following, relating to the [CHANGELOG-UNRELEASED.md](https://github.com/holochain/holochain-rust/blob/develop/CHANGELOG-UNRELEASED.md)

- [x] this is a code change that effects some consumer (e.g. zome developers) of holochain core so it is added to the CHANGELOG-UNRELEASED.md (linked above), with the format `- summary of change [PR#1234](https://github.com/holochain/holochain-rust/pull/1234)`
- [ ] this is not a code change, or doesn't effect anyone outside holochain core development

### CODE REVIEW

File: conductor_api/src/context_builder.rs [hunk 1]
```diff
@@ -109,6 +111,24 @@ impl ContextBuilder {
         Ok(self)
     }
 
+    /// Sets all three storages, chain, DHT and EAV storage, to persistent lmdb based implementations.
+    /// Chain and DHT storages get set to the same pikcle CAS.
```
Comment (zippy) [inline conductor_api/src/context_builder.rs]:
```suggestion
    /// Chain and DHT storages get set to the same CAS.
```

File: crates/conductor_lib/src/context_builder.rs [hunk 2]
```diff
@@ -109,6 +111,24 @@ impl ContextBuilder {
...
+    /// Chain and DHT storages get set to the same pikcle CAS.
+    /// Returns an error if no lmdb storage could be spawned on the given path.
+    pub fn with_lmdb_storage<P: AsRef<Path>>(mut self, path: P, initial_mmap_bytes: Option<usize>) -> Result<Self, HolochainError> {
+        let base_path: PathBuf = path.as_ref().into();
+        let cas_path = base_path.join("cas");
+        let eav_path = base_path.join("eav");
+        fs::create_dir_all(&cas_path)?;
+        fs::create_dir_all(&eav_path)?;
+
+        let file_storage = Arc::new(RwLock::new(LmdbStorage::new(&cas_path, initial_mmap_bytes)));
```
Comment (AshantiMutinta) [inline crates/conductor_lib/src/context_builder.rs]:
```suggestion
        let cas_storage = Arc::new(RwLock::new(LmdbStorage::new(&cas_path, initial_mmap_bytes)));
```

File: crates/conductor_lib/src/conductor/admin.rs [hunk 3]
```diff
@@ -201,13 +201,14 @@ impl ConductorAdmin for Conductor {
...
             agent: agent_id.to_string(),
-            storage: StorageConfiguration::Pickle {
+            storage: StorageConfiguration::Lmdb {
                 path: storage_path
                     .to_str()
                     .ok_or_else(|| {
                         HolochainError::ConfigError(format!("invalid path {:?}", storage_path))
                     })?
                     .into(),
+                initial_mmap_bytes: None,
```
Comment (lucksus) [inline crates/conductor_lib/src/conductor/admin.rs]: @willemolding, the second parameter was missing. I've added this line. Is this correct?

Comment (willemolding) [reply crates/conductor_lib/src/conductor/admin.rs]: Yep I must have missed that one. Looking good

**Pull request merged**